### ISSUE TITLE
quickbrick needs to know about bright vs dark survey

### ISSUE THREAD

**Issue by moustakas** [desihub/desisim#101]
```quickbrick``` is barfing because desimodel/data/desi.yaml specifies ```exptime_dark``` and ```exptime_bright``` whereas ```quickbrick``` only expects ```exptime```:
  https://github.com/desihub/desisim/blob/master/bin/quickbrick#L59

I can fix this by adding a bright vs dark optional input to ```quickbrick```, which would also be an opportunity to include default bright-sky conditions to the simulated output spectra.

**Comment by sbailey** [comment]
`desimodel.io.load_desiparams` has this temporary hack while transitioning codes to use `exptime_dark` and `exptime_bright`:
```
    #- for temporary backwards compability after 'exptime' -> 'exptime_dark'
    if ('exptime' not in _params) and ('exptime_dark' in _params):
        _params['exptime'] = _params['exptime_dark']
```
i.e. quickbrick should work if using the latest desimodel code in addition to the latest desimodel data.

None-the-less, +1 for updating quickbrick to know about bright vs. dark and grabbing the appropriate exposure times, templates, and sky conditions.

**Comment by dkirkby** [comment]
Defining the canonical grey and bright observing sky conditions sounds like a great idea but should be a separate issue (labeled 'enhancement' rather than 'bug').

The root cause of this issue is that we do not have a robust way to synchronize version updates between packages.  In this case, incompatible versions resulted in a crash, which is much better than the alternative of an undetected incorrect result.

The first thing I would try is to define `install_requires` in `setup.py`:
```
install_requires=[
    'desimodel >= 0.4.2' 
]
```
See [here](http://peak.telecommunity.com/DevCenter/setuptools#declaring-dependencies) for details.  There are other package dependencies that should also be listed here if this approach works.

**Comment by sbailey** [comment]
Should this be tracked in setup.py, requirements.txt, both, or have setup.py parse requirements.txt ?  I'm not sure what the recommended practices are here.  If we can chose, I suspect that we'll have better luck with developers keeping a separate requirements.txt file up-to-date rather than setup.py itself.  Let's try to avoid having to keep both in sync.

I see that our current setup.py has:
```
setup_keywords['requires'] = ['Python (>2.7.0)']
# setup_keywords['install_requires'] = ['Python (>2.7.0)']
```
Does Ben or someone else know why the 'install_requires' line is commented out, vs. the 'requires' line?

**Comment by dkirkby** [comment]
I think the short answer is that we need to modify `setup.py` to solve this problem.  The longer answer is [here](http://python-packaging-user-guide.readthedocs.org/en/latest/requirements/).  An even longer answer that makes a distinction between "application" and "reusable" packages (which is `desisim`?) is [here](http://blog.miguelgrinberg.com/post/the-package-dependency-blues).

**Comment by dkirkby** [comment]
Note that `install_requires` dependencies are checked at install time, not run time, so don't protect someone from cloning desisim and running directly from the cloned directory with incompatible dependencies.  However, the checks will run on travis and when developers run `python setup.py develop / install` or `desiInstall`.

The next question is what happens when a dependency requirement is not met.  The default setuptools assumption is that the necessary version is registered with PyPI, but there are [work arounds](https://pythonhosted.org/setuptools/setuptools.html#dependencies-that-aren-t-in-pypi).

**Comment by moustakas** [comment]
Closed in #114.
